Windows doc update (#185)
harvester/docs: docs/vm/create-windows-vm.md

    1. Click on `Load driver` and click `Browse` on the dialog box, and find a CD-ROM drive with `VMDP-WIN` prefix. Find the driver directory according to the Windows version you're installing, for example: Windows Server 2012r2 should expand `win8.1-2012r2` and choose the `pvvx` directory inside.
    ![find-virtio-driver-directory](assets/find-virtio-driver-directory.png)
    2. Click `OK` to allow installer to scan this directory for drivers, choose `SUSE Block Driver for Windows` and clock `Next` to load the driver.
    ![select-virtio-block-driver](assets/select-virtio-block-driver.png)
    3. Wait for the installer to load up the driver. If you're choosing correct version of driver the `virtio` volumes should be detected once the driver is loaded.
    ![installer-found-virtio-drive](assets/installer-found-virtio-drive.png)
    4. Operate this drive at your own discretion.

4. [Optional] If you are using other `virtio` based hardware like network adapter, you will need to manually install those drivers after completed the installation. To install drivers, open VMDP driver disk, and use the installer based on your platform.

> The support matrix of VMDP driver pack for Windows are as follows (assumes VMDP CD-ROM drive letter are E):

Reviewer: ```suggestion
The support matrix of the VMDP driver pack for Windows are as follows (assume the VMDP CD-ROM drive path is E):
```

| Version | Supported | Driver path |
| :-----: | :-------: | :---------- |
| Windows 7 | No | `N/A` |
| Windows Server 2008 | No | `N/A` |
| Windows Server 2008r2 | No | `N/A` |
| Windows 8 x86(x64) | Yes | `E:\win8-2012\x86(x64)\pvvx` |
| Windows Server 2012 x86(x64) | Yes | `E:\win8-2012\x86(x64)\pvvx` |
| Windows 8.1 x86(x64) | Yes | `E:\win8.1-2012r2\x86(x64)\pvvx` |
| Windows Server 2012r2 x86(x64) | Yes | `E:\win8.1-2012r2\x86(x64)\pvvx` |
| Windows 10 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows Server 2016 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows Server 2019 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows 11 x86(x64) | Yes | `E:\win10-2004\x86(x64)\pvvx` |
| Windows Server 2022 x86(x64) | Yes | `E:\win10-2004\x86(x64)\pvvx` |

!!! note

>     If you need to use other VM template rather than the `windows-iso-image-base-template` we provided, and you still need `virtio` devices, please be sure to provide your own virtio driver disk in order to detect hardware correctly.

Reviewer: ```suggestion
    If you didn't use the `windows-iso-image-base-template` template, and you still need `virtio` devices, please make sure to add your custom Windows virtio driver to allow it to detect the hardware correctly.
```

## Known Issues

### Windows ISO unable to boot when using EFI mode

When using EFI mode with Windows, you may find out the system booted with other devices like HDD or UEFI shell like the one below:

![efi-shell](assets/efi-shell.png)

> That's because Windows will prompt a `Press any key to boot from CD or DVD...` to let user decide whether to boot to installer ISO or not, and it needs human interaction to allow boot from CD or DVD.

Reviewer: ```suggestion
That's because Windows will prompt a `Press any key to boot from CD or DVD...` to let the user decide whether to boot from the installer ISO or not, and it needs human interaction to allow boot from CD or DVD.
```

![boot-from-cd](assets/boot-from-cd.png)

As a workaround, if the system already booted into UEFI shell, you can type in `reset` to let the system reboot, once the prompt appears you can press any key to let system boot from Windows ISO.

### VM crashes when reserved memory not enough

> There is a known issue with Windows VM, when allocated more than 8GiB of memory to Windows VM and not enough reserved memory configured, the VM crashes without warning. This can be temporarily fixed by allocating at least 256MiB of reserved memory to the VM before the installation. 

Reviewer: ```suggestion
There is a known issue with Windows VM, when allocated more than 8GiB of memory to Windows VM and not enough reserved memory configured, the VM crashes without warning. This can be fixed by allocating at least 256MiB of reserved memory to the VM on the advanced tab. 
```

Click `Edit config` of your crashing VM, go to the `Advanced Options` tab, click `Show More` and fill in at least 256MiB of reserved memory to prevent crashes.

![reserved-memory-config](assets/reserved-memory-config.png)

We will add a default 256MiB of reserved memory to the Windows template to prevent this problem in the future release.

### BSoD (Blue Screen of Death) at first boot time of Windows

There is a known issue with Windows VM using Windows Server 2016 and above, a BSoD with error code `KMODE_EXCEPTION_NOT_HANDLED` may appears at the first boot time of Windows. We are still looking into it and will fix this issue in the future release.

As a workaround, you can create or modify the file `/etc/modprobe.d/kvm.conf` within the installation of Harvester by updating `/oem/99_custom.yaml` like below:

```YAML
name: Harvester Configuration
stages:
  initramfs:
  - commands: # ...
    files:
    - path: /etc/modprobe.d/kvm.conf
      permissions: 384
      owner: 0
      group: 0
      content: |
          options kvm ignore_msrs=1
      encoding: ""
      ownerstring: ""
      # ...
```
!!! note

>     This is still a temporary solution and it's danger to modify this configuration. We don't recommend user to do this. For more information please refer to [this issue](https://github.com/harvester/harvester/issues/276).

Reviewer: ```suggestion
    This is still an experimental solution. For more information, please refer to [this issue](https://github.com/harvester/harvester/issues/276), and please let us know if you have encountered any issues after applying this workaround.
```

!!! note
    If you need to add advanced settings, you can edit the VM configuration directly by clicking on `Edit as YAML`. 
    And if you want to cancel all changes made, click `Cancel`.

## Installation of Windows

1. Select the VM you just created, and click `Start` to boot up the VM.(If you checked `Start virtual machine on creation` the VM will start automatically once it's created)

> 2. Boot into the installer, and follow the instruction given by the installer.

Reviewer 2: ```
2. Boot into the installer, and follow the instructions given by the installer.
```

The support matrix of VMDP driver pack for Windows are as follows (assume the VMDP CD-ROM drive path is E):

| Version | Supported | Driver path |
| :-----: | :-------: | :---------- |
| Windows 7 | No | `N/A` |
| Windows Server 2008 | No | `N/A` |
| Windows Server 2008r2 | No | `N/A` |
| Windows 8 x86(x64) | Yes | `E:\win8-2012\x86(x64)\pvvx` |
| Windows Server 2012 x86(x64) | Yes | `E:\win8-2012\x86(x64)\pvvx` |
| Windows 8.1 x86(x64) | Yes | `E:\win8.1-2012r2\x86(x64)\pvvx` |
| Windows Server 2012r2 x86(x64) | Yes | `E:\win8.1-2012r2\x86(x64)\pvvx` |
| Windows 10 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows Server 2016 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows Server 2019 x86(x64) | Yes | `E:\win10-server\x86(x64)\pvvx` |
| Windows 11 x86(x64) | Yes | `E:\win10-2004\x86(x64)\pvvx` |
| Windows Server 2022 x86(x64) | Yes | `E:\win10-2004\x86(x64)\pvvx` |

!!! note
    If you didn't use the `windows-iso-image-base-template` template, and you still need `virtio` devices, please make sure to add your custom Windows virtio driver to allow it to detect the hardware correctly.

## Known Issues

### Windows ISO unable to boot when using EFI mode

> When using EFI mode with Windows, you may find out the system booted with other devices like HDD or UEFI shell like the one below:

Reviewer 2: ```
When using EFI mode with Windows, you may find the system booted with other devices like HDD or UEFI shell like the one below:
```

![efi-shell](assets/efi-shell.png)

> That's because Windows will prompt a `Press any key to boot from CD or DVD...` to let the user decide whether to boot from the installer ISO or not, and it needs human interaction to allow boot from CD or DVD.

Reviewer 2: ```
That's because Windows will prompt a `Press any key to boot from CD or DVD...` to let the user decide whether to boot from the installer ISO or not, and it needs human intervention to allow the system to boot from CD or DVD.
```

![boot-from-cd](assets/boot-from-cd.png)

> As a workaround, if the system already booted into UEFI shell, you can type in `reset` to let the system reboot, once the prompt appears you can press any key to let system boot from Windows ISO.

Reviewer 2: ```
Alternately if the system has already booted into the UEFI shell, you can type in `reset` to force the system to reboot again. Once the prompt appears you can press any key to let system boot from Windows ISO.
```

### VM crashes when reserved memory not enough

> There is a known issue with Windows VM, when allocated more than 8GiB of memory to Windows VM and not enough reserved memory configured, the VM crashes without warning. This can be fixed by allocating at least 256MiB of reserved memory to the on the `Advanced Options` tab. 

Reviewer 2: There is a known issue with Windows VM when it is allocated more than 8GiB of without enough reserved memory configured. The VM crashes without warning.

Reviewer 2: This can be fixed by allocating at least 256MiB of reserved memory to the template on the `Advanced Options` tab.
